### PR TITLE
Dont throw any errors when receiving messages

### PR DESCRIPTION
**Pull request by emirkin**
Don't throw any errors when receiving messages otherwise the queues will get flooded

### CODE REVIEW

File: lib/pub_sub/message.rb [hunk 1]
```diff
@@ -13,18 +13,17 @@ def initialize(payload)
     def process
       begin
         validate_message!
-      rescue PubSub::ServiceUnknown => e
-        return
+        handler.process(data)
```
Comment (ashoda) [inline lib/pub_sub/message.rb]: Not clear what's expected here? You'd like `validate_message!` to throw an error, but for `handler.process` not to?

Comment (emirkin) [inline lib/pub_sub/message.rb]: We will log the error but not fail the parent `process`

Comment (ashoda) [inline lib/pub_sub/message.rb]: Nvm, I see. You're catching all errors instead. What's the overall impact of this change. Will we lost the messages permanently if `handler.process` errors out?

File: spec/lib/pub_sub/message_spec.rb [hunk 2]
```diff
@@ -64,27 +64,29 @@ class EntityUpdate
...
-      let(:payload) { raw_sns_message['body'] }
-      it 'with exception' do
+      it 'on unknown types' do
         allow(PubSub.config.subscriptions).to receive(:[]).with(
           'entity-service-prod'
         ).and_return(['unknown_type'])
-        expect{subject.process}.to raise_error(PubSub::MessageTypeUnknown)
+        allow(PubSub.config).to receive_message_chain("logger.error").and_return(anything())
+        # first, make sure the validator throws an exception
+        expect{subject.validate_message!}.to raise_error(PubSub::MessageTypeUnknown)
```
Comment (ashoda) [inline spec/lib/pub_sub/message_spec.rb]: Might think about pulling out specs around `validate_message!` and `process` into their own unit tests. I realize that there's overlaps concerning exceptions but this spec seems to have bloated. It's testing multiple methods/behaviors in one place.

File: lib/pub_sub/message.rb (hunk 1 above)
Comment (emirkin) [inline lib/pub_sub/message.rb]: Check out the latest commit

File: spec/lib/pub_sub/message_spec.rb (hunk 2 above)
Comment (emirkin) [inline spec/lib/pub_sub/message_spec.rb]: At the moment, all this logic is inter-connected, so testing one without the other is of little use. I'll keep things as is for now but will make a mental note to possibly revisit later

**Pull request merged**